### PR TITLE
Request confirmation email resending

### PR DESCRIPTION
**Pull request by Devessier**

### CODE REVIEW

File: packages/server/app/Controllers/Http/AuthenticationController.ts [hunk 1]
```diff
@@ -247,6 +248,8 @@ export default class AuthenticationController {
             'resendConfirmationEmail',
         );
         if (hasReachedRateLimit === true) {
+            response.status(429);
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/AuthenticationController.ts:251]: Http status code, just like english irregular verbs should be learned at school !

File: packages/client/machines/appMachine.ts [hunk 2]
```diff
@@ -453,6 +456,50 @@ export function createAppMachine({
                             },
                         },
 
+                        resendingConfirmationEmail: {
```
Comment (prastoin) [inline packages/client/machines/appMachine.ts:459]: Thoughts: This means that even if the user has confirmed her email or user is not on email confirmation screen he could ask for email resend ?

File: packages/client/__tests__/AuthenticationResendingConfirmationEmail.test.tsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+import { withAuthentication } from '../tests/server/handlers';
+import { db, generateAuthenticationUser } from '../tests/data';
+
+interface TestingContext {
+    screen: ReturnType<typeof render>;
+}
+
+const existingUser = generateAuthenticationUser();
+
+/**
```
Comment (prastoin) [inline packages/client/__tests__/AuthenticationResendingConfirmationEmail.test.tsx:25]: Praise: great !

**Comment by prastoin** [comment]
Great implementation ! waiting for CI to be fixed before approving and merging !

File: packages/client/machines/appMachine.ts (hunk 2 above)
Comment (Devessier) [reply packages/client/machines/appMachine.ts:459]: `resendingConfirmationEmail` state is at the same level  as `formHandler` and `pollingHandler` states inside `waitingForUserEmailConfirmation` state. I think what you describe can not happen 🤔 Do I miss something?

File: packages/server/app/Controllers/Http/AuthenticationController.ts (hunk 1 above)
Comment (Devessier) [reply packages/server/app/Controllers/Http/AuthenticationController.ts:251]: Lol, actually Github Copilot suggested me the status code. 😆

File: packages/client/machines/appMachine.ts (hunk 2 above)
Comment (prastoin) [reply packages/client/machines/appMachine.ts:459]: Ah my bad !

**Review by prastoin — APPROVED** [review]
Ready to be merged !

**Pull request merged**